node/node.go

```diff
@@ -1493,7 +1515,16 @@ func (app *App) startServices(ctx context.Context) error {
 	app.blockGen.Start(ctx)
 	app.certifier.Start(ctx)
 	app.eg.Go(func() error {
-		return app.proposalBuilder.Run(ctx)
+		if app.proposalBuilder != nil {
+			return app.proposalBuilder.Run(ctx)
+		}
+		return nil
+	})
+	app.eg.Go(func() error {
+		if app.remoteProposalBuilder != nil {
+			return app.remoteProposalBuilder.Run(ctx)
+		}
+		return nil
 	})
```

Comment:
You could avoid running the goroutine if not needed:
```suggestion
	if app.remoteProposalBuilder != nil {
		app.eg.Go(func() error {
			return app.remoteProposalBuilder.Run(ctx)
		})
	}
```

Same above.